### PR TITLE
P1614R2 Spaceship: Clause 27: Time (partial, pre-C++20 types)

### PR DESCRIPTION
**Pull request by statementreply**
This PR patches pre-C++20 `<chrono>` types.

#323 has already implemented WG21-P1614 for the new calendar types.

### CODE REVIEW

**Comment by MattStephanson** [comment]
I had implemented these for `utc_clock` et al. I don't mind waiting, just FYI that this is blocking other work.

**Comment by statementreply** [comment]
> I had implemented these for `utc_clock` et al. I don't mind waiting, just FYI that this is blocking other work.

Would it be better if I retarget this PR to main branch to unblock `<chrono>` work?

**Comment by MattStephanson** [comment]
Oh, I forgot there's a feature branch for these. I'm using `operator<=>(leap_second,sys_time` (thus, `operator<=>(time_point,time_point)`) for `std::get_leap_second_info`, so it would be useful for me to have it. But I could also rewrite it and move the leap-related `<=>` to `feature/spaceship` as well.

**Comment by statementreply** [comment]
Force-pushed, retarget to `main`.

File: tests/std/tests/P1614R2_spaceship/env.lst
Comment (StephanTLavavej) [inline tests/std/tests/P1614R2_spaceship/env.lst:3]: This new test needs to be mentioned in `test.lst`, or it won't run in the MSVC-internal repo.

**Review by StephanTLavavej — COMMENTED** [review]
Looks great! I'll push a `test.lst` update.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I've mirrored this to internal MSVC-PR-305171.

Please note: Further changes are still possible (e.g. if you notice something, or if final review leads to more feedback that should be addressed before merging), but they must be replicated to the mirror PR in order to avoid codebase divergence.

**Review by MahmoudGSaleh — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks for implementing these timeships and unblocking further work on C++20 timezones! :timer_clock: :flying_saucer: :smiley_cat:

**Pull request merged**
